src/swap/SwapTransactionDetails.tsx

```diff
@@ -263,7 +295,196 @@ export function SwapTransactionDetails({
   )
 }
 
+const useTotalSwapFeesInLocalCurrency = ({
+  estimatedNetworkFee,
+  networkFeeTokenId,
+  appFee,
+  estimatedCrossChainFee,
+}: {
+  estimatedNetworkFee?: BigNumber
+  networkFeeTokenId: string
+  appFee?: {
+    amount: BigNumber
+    token: TokenBalance
+    percentage: BigNumber
+  }
+  estimatedCrossChainFee?: BigNumber
+}) => {
+  const networkFeeTokenInfo = useTokenInfo(networkFeeTokenId)
+  const nativeTokenInfo = useTokenInfo(
+    networkFeeTokenInfo ? getTokenId(networkFeeTokenInfo.networkId) : undefined
+  )
+  const usdToLocalRate = useSelector(usdToLocalCurrencyRateSelector)
+
+  if (!usdToLocalRate) {
+    Logger.warn(
+      'swap/useTotalSwapFeesInLocalCurrency',
+      'Cannot calculate swap fees due to missing usdToLocalRate'
+    )
+    return null
+  }
+
+  if (!estimatedNetworkFee) {
+    Logger.warn(
+      'swap/useTotalSwapFeesInLocalCurrency',
+      `Cannot calculate swap fees due to missing estimatedNetworkFee`
+    )
+    return null
+  }
+
+  if (!networkFeeTokenInfo?.priceUsd) {
+    Logger.warn(
+      'swap/useTotalSwapFeesInLocalCurrency',
+      `Cannot calculate swap fees due to missing usd price for network fee token ${networkFeeTokenId}`
+    )
+    return null
+  }
+
+  if (appFee && appFee.amount.gt(0) && !appFee.token.priceUsd) {
+    Logger.warn(
+      'swap/useTotalSwapFeesInLocalCurrency',
+      `Cannot calculate swap fees due to missing usd price for app fee token ${appFee.token.tokenId}`
+    )
+    return null
+  }
```

Comment:
Though the other cases can't really happen (we should always have a price for the network fee and cross-chain fee), this one for sure happens.
Should we count it as 0 then to at least partially show the "total"?

Reply:
my concern is that we're only showing this "bite size" version of the fee in fiat, if we don't have the price of some tokens in the fee and set them to 0, then the user could see a low fee value and proceed with the swap thinking it's a great deal. my intention was to show only the total fee if we have the complete picture, if not then don't show anything and require the user to go into the details screen, where they can see the breakdown of the fees to understand the full picture.